src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+// computer algrithm for hard level computer
+//
+// features:
+// - Count cards on the table, record all the played card                                               (x)
+// - Quickly get rid of matched numbers in the hand.                                                    (x)
+// - Keeping at most one Wild Card until the very end.                                                  (x)
+// - use reverse card when you have no card to play                                                     (x)
+// - Play series of similar color cards first, use draw 2 card,                                         (x)
+//   skip card, or reverse card first based on the most color that computer has                         (x)
+// - When playing a Wild Card +4, always change the color based on what the computer has the most       (x)
+// - Try to keep at least one +2 card in your hand.         (x)                                         (x)
+// - Play higher number cards first to get more points      (x)                                         (x)
+// - Note the last card color when players have one card left and avoid to play it.
+//
+// input :
+// - card on the table (for update and play)
+// - if player "UNO"
+// - card drawed from the deck
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "cards_management.h"
+ 
+#define MAX 40
+#define hand_MAX 20
+#define plarCardNumber 10
+
+///* (not used for now) */
+//struct CARD record_array[MAX]; // array for recording all the cards
+//int record_size = 0, handcard_size = 0;
+//
+//void record(struct CARD card)
+//{
+//    record_size++;
+//    record_array[record_size - 1] = card;
+//}
+
+/* 
+* @brief: function that find the most occurrence with the color 
+* 
+* @return CardColor the color with moset occurrence
+*/
+enum CardColor  find_most_color(struct DECK *cards_on_hand)
+{
+    int R_count = 0, G_count = 0, B_count = 0, Y_count = 0, most = 0, least = 0;
+    struct DECK *temp;
+    struct DECK *head = cards_on_hand;
+    if (head == NULL)
+    {
+        printf("hand card is empty!");
+        return 0;
+    }
+    temp = head;
+    while (temp != NULL)
+    {
+        if (temp->card.color == RED)
+        {
+            R_count++;
+        }
+        else if (temp->card.color == BLUE)
+        {
+            B_count++;
+        }
+        else if (temp->card.color == GREEN)
+        {
+            G_count++;
+        }
+        else
```

Comment:
It's better to check else if (temp->card.color == YELLOW), just in case there is invalid value for temp->card.color.